### PR TITLE
Add some informations about the Dynamic.js view

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: doc/integrator/ngeo.rst [hunk 1]
```diff
@@ -108,3 +108,23 @@ The map configuration will be here:
    If you previously had a CGXP application and want to keep your permalinks
    compatible, the resolutions should be the same as in the previous
    application.
+
+Dynamic.js view
+---------------
+
+To configure the ngeo constance with dynamic or configurable bales we have the dynamic view.
```
Comment (arnaud-morvan) [inline doc/integrator/ngeo.rst]: constance => constants

Comment (arnaud-morvan) [reply doc/integrator/ngeo.rst]: bales ?

File: doc/integrator/ngeo.rst [hunk 2]
```diff
@@ -108,3 +108,23 @@ The map configuration will be here:
...
+
+This view is configurable in the vars files in the section ``interfaces_config``.
+The sub section is the interface name, and after we have:
+
+* ``redirect_interface``: interface on witch one we can be redirected if we use the wrong type of device
+  (mobile/desktop).
+* ``do_redirect``: directly to the redirect.
+* ``constants``: Dirrectly define an ``angular`` constant in the vars file.
+* ``routes``: key: constant name, value name of the route witch one we want to have the URL.
```
Comment (arnaud-morvan) [inline doc/integrator/ngeo.rst]: I do not understand.
How I suppose it works: 
``routes``: ngeo constant name: pyramid route name, same as a constant, but interpreted on server side through pyramid request.route_url.

File: doc/integrator/ngeo.rst [hunk 3]
```diff
@@ -108,3 +108,23 @@ The map configuration will be here:
...
+* ``static``: key: constant name, value name of the static view that we want to have the URL.
```
Comment (arnaud-morvan) [inline doc/integrator/ngeo.rst]: Same a routes.

File: doc/integrator/ngeo.rst (hunk 1 above)
Comment (sbrunner) [reply doc/integrator/ngeo.rst]: I don't understand you second message...

File: doc/integrator/ngeo.rst (hunk 2 above)
Comment (sbrunner) [reply doc/integrator/ngeo.rst]: Yes :-)

File: doc/integrator/ngeo.rst (hunk 3 above)
Comment (sbrunner) [reply doc/integrator/ngeo.rst]: Yes :-)

Comment (sbrunner) [reply doc/integrator/ngeo.rst]: Is it OK or can we do something better?

File: doc/integrator/ngeo.rst (hunk 2 above)
Comment (arnaud-morvan) [reply doc/integrator/ngeo.rst]: "configuration bales" seems stange to me, but I'm not high level english speaker.

File: doc/integrator/ngeo.rst (hunk 1 above)
Comment (sbrunner) [reply doc/integrator/ngeo.rst]: I just understand, fixed, thanks :-)

File: doc/integrator/ngeo.rst [hunk 4]
```diff
@@ -108,3 +108,23 @@ The map configuration will be here:
...
+
+To configure the ngeo constance with dynamic or configurable bales we have the dynamic view.
+
+This view is configurable in the vars files in the section ``interfaces_config``.
+The sub section is the interface name, and after we have:
+
+* ``redirect_interface``: interface on witch one we can be redirected if we use the wrong type of device
+  (mobile/desktop).
+* ``do_redirect``: directly to the redirect.
+* ``constants``: Dirrectly define an ``angular`` constant in the vars file.
```
Comment (arnaud-morvan) [inline doc/integrator/ngeo.rst]: Dirrectly => Directly

File: doc/integrator/ngeo.rst [hunk 5]
```diff
@@ -108,3 +108,23 @@ The map configuration will be here:
...
+---------------
+
+To configure the ngeo constants with dynamic or configurable values we have the dynamic view.
+
+This view is configurable in the vars files in the section ``interfaces_config``.
+The sub section is the interface name, and after we have:
+
+* ``redirect_interface``: interface on witch one we can be redirected if we use the wrong type of device
+  (mobile/desktop).
+* ``do_redirect``: directly to the redirect.
```
Comment (arnaud-morvan) [inline doc/integrator/ngeo.rst]: to => do ?

File: doc/integrator/ngeo.rst [hunk 6]
```diff
@@ -108,3 +108,23 @@ The map configuration will be here:
...
+
+Dynamic.js view
+---------------
+
+To configure the ngeo constants with dynamic or configurable values we have the dynamic view.
+
+This view is configurable in the vars files in the section ``interfaces_config``.
+The sub section is the interface name, and after we have:
+
+* ``redirect_interface``: interface on witch one we can be redirected if we use the wrong type of device
```
Comment (jwkaltz) [inline doc/integrator/ngeo.rst]: suggestion:
interface to be redirected to if an unexpected device type is used (mobile/desktop)

File: doc/integrator/ngeo.rst [hunk 7]
```diff
@@ -108,3 +108,23 @@ The map configuration will be here:
...
+  (mobile/desktop).
+* ``do_redirect``: directly do the redirect.
+* ``constants``: Directly define an ``angular`` constant in the vars file.
+* ``routes``: key: constant name, value: name of the route witch one we want to have the URL.
```
Comment (jwkaltz) [inline doc/integrator/ngeo.rst]: value description is not clear to me - I am not sure what we are trying to say here?

File: doc/integrator/ngeo.rst [hunk 8]
```diff
@@ -108,3 +108,23 @@ The map configuration will be here:
...
+* ``static``: key: constant name, value: name of the static view that we want to have the URL.
+* ``fulltextsearch_params``: Additional parameters to the full text search view, see
+  :ref:`developer_webservices_fts` for more information.
+* ``wfs_permalink``: Additional values for the ``ngeoWfsPermalinkOptions`` constant, see:
```
Comment (jwkaltz) [inline doc/integrator/ngeo.rst]: ngeoWfsPermalinkOptions is not a constant, but something else, I guess?

File: doc/integrator/ngeo.rst (hunk 7 above)
Comment (sbrunner) [reply doc/integrator/ngeo.rst]: Name of the route for which the URL is wanted. ?

File: doc/integrator/ngeo.rst (hunk 8 above)
Comment (sbrunner) [reply doc/integrator/ngeo.rst]: it's an angular constant name...

Comment (jwkaltz) [reply doc/integrator/ngeo.rst]: But isn't it strange to provide additional values for a constant? (constant = a constant value to me, not something where you can provide additional values)

File: doc/integrator/ngeo.rst (hunk 7 above)
Comment (jwkaltz) [reply doc/integrator/ngeo.rst]: Yes, that sounds good

File: doc/integrator/ngeo.rst (hunk 8 above)
Comment (sbrunner) [reply doc/integrator/ngeo.rst]: In code I do that (in mako template):
```
<%
wfs_permalink = dict(url=request.route_url('mapserverproxy'))
wfs_permalink.update(interface_config.get('wfs_permalink', {}))
%>
module.constant('ngeoWfsPermalinkOptions', ${json.dumps(wfs_permalink) | n});
```
...

Comment (jwkaltz) [reply doc/integrator/ngeo.rst]: OK, I understand, thanks.

**Pull request merged**